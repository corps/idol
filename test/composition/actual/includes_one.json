{"includes_one":{"module_name":"includes_one","types_by_name":{"IncludesOneType":{"fields":{},"is_a":{"literal":null,"primitive_type":"int","reference":{"module_name":"","qualified_name":"","type_name":""},"struct_kind":"Scalar"},"named":{"module_name":"includes_one","qualified_name":"includes_one.IncludesOneType","type_name":"IncludesOneType"},"options":[],"tags":[]}},"types_dependency_ordering":["IncludesOneType"]},"common.type_structs":{"module_name":"common.type_structs","types_by_name":{"AnyMap":{"fields":{},"is_a":{"literal":null,"primitive_type":"any","reference":{"module_name":"","qualified_name":"","type_name":""},"struct_kind":"Map"},"named":{"module_name":"common.type_structs","qualified_name":"common.type_structs.AnyMap","type_name":"AnyMap"},"options":[],"tags":[]},"Int":{"fields":{},"is_a":{"literal":null,"primitive_type":"int","reference":{"module_name":"","qualified_name":"","type_name":""},"struct_kind":"Scalar"},"named":{"module_name":"common.type_structs","qualified_name":"common.type_structs.Int","type_name":"Int"},"options":[],"tags":[]},"Str":{"fields":{},"is_a":{"literal":null,"primitive_type":"string","reference":{"module_name":"","qualified_name":"","type_name":""},"struct_kind":"Scalar"},"named":{"module_name":"common.type_structs","qualified_name":"common.type_structs.Str","type_name":"Str"},"options":[],"tags":[]},"Bools":{"fields":{},"is_a":{"literal":null,"primitive_type":"bool","reference":{"module_name":"","qualified_name":"","type_name":""},"struct_kind":"Repeated"},"named":{"module_name":"common.type_structs","qualified_name":"common.type_structs.Bools","type_name":"Bools"},"options":[],"tags":[]},"Doubles":{"fields":{},"is_a":{"literal":null,"primitive_type":"int","reference":{"module_name":"common.type_structs","qualified_name":"common.type_structs.Double","type_name":"Double"},"struct_kind":"Repeated"},"named":{"module_name":"common.type_structs","qualified_name":"common.type_structs.Doubles","type_name":"Doubles"},"options":[],"tags":[]},"Double":{"fields":{},"is_a":{"literal":null,"primitive_type":"double","reference":{"module_name":"","qualified_name":"","type_name":""},"struct_kind":"Scalar"},"named":{"module_name":"common.type_structs","qualified_name":"common.type_structs.Double","type_name":"Double"},"options":[],"tags":[]}},"types_dependency_ordering":["Str","Int","Double","Doubles","Bools","AnyMap"]}}
