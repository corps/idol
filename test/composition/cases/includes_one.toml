[[includes]]
from = "common.type_structs"

[IncludesOneType]
is_a = "int"