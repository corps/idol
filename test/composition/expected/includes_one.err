Error processing modules: is_a declaration Double[]: Error parsing primitive type: unknown variant `Double`, expected one of `int`, `double`, `string`, `bool`, `any`
Error: 1
