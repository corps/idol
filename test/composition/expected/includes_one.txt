=== module aliases.type_structs
aliases.type_structs.Str: string # 
aliases.type_structs.Int: int # 
aliases.type_structs.Double: double # 
aliases.type_structs.Doubles: aliases.type_structs.Double[] # 
aliases.type_structs.Bools: bool[] # 
aliases.type_structs.AnyMap: any{} # 
=== module includes_one
includes_one.IncludesOneType: int # 
